.github/workflows/benchmark-pg_search.yml

```diff
@@ -105,13 +101,6 @@ jobs:
         working-directory: pg_search/
         run: cargo paradedb bench eslogs query-search-index --url postgresql://localhost:288${{ matrix.pg_version }}/postgres
 
-      - name: Save criterion output, if dev
-        if: steps.check_skip.outputs.skip_remaining_steps != 'true' && github.ref == 'refs/heads/dev'
-        uses: actions/cache/save@v4
-        with:
-          path: target/criterion
-          key: "pg_search-criterion-benchmark"
-
```

Comment:
I don't think removing this is correct.

The goal here was for the `dev` branch to be the only place where the criterion files are saved, so that PRs have a baseline to compare against.  

If all PRs start caching their criterion output, then we won't know what any given PR is actually comparing against.

Reply:
The issue is the benchmarks weren't saving at all, because it can't override the cache key when done like this... Hmm let me see if we can come up with a different solution

Reply:
Okay, I have a good one. I will submit a PR to this PR.